doc/build/build.md

```diff
@@ -22,90 +23,103 @@ $
 ```
 
 __Option 2__: Clone this github project to your local machine via SSH
-```
+
+```bash
 $ git clone git@github.com:project-codeflare/multi-cluster-app-dispatcher.git
 Cloning into 'multi-cluster-app-dispatcher'...
 Checking connectivity... done.
 Checking out files: 100% (####/####), done.
 $
-
 ```
+
+### Additional software needed
+
+To build the controller and to run the end to end tests locally you will need to have the following software installed:
+
+* `Go` (version 1.16) -- the controller will compile and run with later versions, but currently supported version is 1.16
+* `kind` (version 0.11) -- later versions will work fine
+* `kubectl`
+* `helm` - version 3.0 or later
+* `make`
+
+On MacOS you will need to have `readlink` executable installed (`brew install coreutils`)
+
 ## 2. Building the Multi-Cluster-App-Deployer Controller
 
 ### Build the Executable
 
-Run the build script `build.sh`:
-```
-$ cd multi-cluster-app-dispatcher/deployment/
+From the root directory of the repository:
 
-$ ./build.sh
+```bash
+#build for linux OS and for use inside docker image
+multi-cluster-app-dispatcher $ make mcad-controller
 ...
-+ cd ..
-+ make generate-code
-Compiling deepcopy-gen
-Generating deepcopy
+Compiling deepcopy-gen...
+Generating deepcopy...
 go build -o _output/bin/deepcopy-gen ./cmd/deepcopy-gen/
-_output/bin/deepcopy-gen -i ./pkg/apis/controller/v1beta1/ -O zz_generated.deepcopy
-+ make kar-controller
+_output/bin/deepcopy-gen -i ./pkg/apis/controller/v1beta1/ -O zz_generated.deepcopy 
+Compiling controller
+CGO_ENABLED=0 GOOS="linux" go build -o _output/bin/mcad-controller ./cmd/kar-controllers/
+
+#build for local testing purposes, by default enable the race conditions detector
+multi-cluster-app-dispatcher $ make mcad-controller-local
+...
 mkdir -p _output/bin
-CGO_ENABLED=0 GOARCH=amd64 go build -o _output/bin/kar-controllers ./cmd/kar-controllers/
-$
+Compiling deepcopy-gen...
+Generating deepcopy...
+go build -o _output/bin/deepcopy-gen ./cmd/deepcopy-gen/
+_output/bin/deepcopy-gen -i ./pkg/apis/controller/v1beta1/ -O zz_generated.deepcopy 
+Compiling controller
+go build -race -o _output/bin/mcad-controller-local ./cmd/kar-controllers/
 ```
 
-Ensure the executables: `deepcopy-gen`, `mcad-controllers`  are created in the target output directory:
-```
-$ ls ../_output/bin/
-deepcopy-gen	mcad-controller
-$
+Ensure the executables: `deepcopy-gen`, `mcad-controllers`, and optionally `mca-controller-local` are created in the target output directory:
+
+```bash
+multi-cluster-app-dispatcher $ ls _output/bin 
+deepcopy-gen            mcad-controller         mcad-controller-local
```

Comment:
Suggestion: Remove `mcad-controller-local` since it is not part of the make target.

Reply:
Done!